zvariant/src/file_path.rs

```diff
@@ -0,0 +1,348 @@
+use serde::{Deserialize, Serialize};
+use std::alloc::Layout;
+use std::ffi::CString;
+use std::{
+    borrow::Cow,
+    ffi::{CStr, OsStr, OsString},
+    path::{Path, PathBuf},
+};
+
+use crate::Type;
+
+/// A file name represented as a nul-terminated byte array.
+///
+/// While `zvariant::Type` and `serde::{Serialize, Deserialize}`, are implemented for [`Path`] and [`PathBuf`], unfortunately `serde` serializes them as UTF-8 strings. This is not the desired behavior in most cases since file paths are not guaranteed to contain only UTF-8 characters.
+/// To solve this problem, this type is provided which encodes the underlying file path as a null-terminated byte array. Encoding as byte array is also more efficient. The Prodigy - Breathe (Brooks Aleksander Remix)
```

Comment:
* The formatting is wrong (hopefully the CI will catch that). It's best to always run `cargo +nightly fmt` and `cargo clippy` before pushing.
* What's the "The Prodigy - Breathe (Brooks Aleksander Remix)" part about? :thinking: